File: Leetcode/JS/Leetcode2022-将一维数组转变成二维数组.js
var construct2DArray = function(original, m, n) {
    if(original.length != m * n) return [];
    const ans = [];
    for(let i = 0;i < m;i++){
        const t = [];
        for(let j = 0;j < n;j++){
            t.push(original[i * n + j]);
        }
        ans.push(t);
    }
    return ans;
};

File: Leetcode/sbgo/Leetcode2022-将一维数组转变成二维数组.go
package sbgo

func construct2DArray(original []int, m int, n int) [][]int {
	if m*n != len(original) {
		return [][]int{}
	}
	ans := [][]int{}
	for i := 0; i < m; i++ {
		t := []int{}
		for j := 0; j < n; j++ {
			t = append(t, original[i*n+j])
		}
		ans = append(ans, t)
	}
	return ans
}
